### DISCUSSION TITLE
No local DNS working without internet connection

### DISCUSSION THREAD

**Discussion by Callahan78** [AdguardTeam/AdGuardHome#3575] [Q&A]
I have the following (DNS) setup:

Clients > AdGuard Home (Ubuntu)> Microsoft DNS > Internet

All my clients are using AdGuard as DNS server
AdGuard uses a Microsoft DNS as upstream.
Microsoft does DNS for local adresses and internet addresses.

Everytime when the internet connection is lost, the internal DNS resolution does no more work. So all my devices couldn´t resolve the other local addresses.
When I bypass AdGuard and use MS DNS directly, it´s working.

Long story short: AdGuard does not work (asking the upstream DNS for local addresses), when there is no internet connection.

Does anyone have a clue, how to fix this?

**Comment by ainar-g** [comment]
Hello.  This sounds a lot like this known issue: #2657.

**Reply by Callahan78** [reply]
Looks exactly the same . . .  known bug since some months and no fix so far?
So the only workaround is to switch to edge? Is that still current today? Last post is from July.

**Reply by ainar-g** [reply]
Yes, since the issue is still open, it is currently unresolved.  It is planned to be resolved in the upcoming v0.107.0 release, but I can't give any precise time frames.
